internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -335,49 +335,126 @@ func resourceDatabricksWorkspace() *pluginsdk.Resource {
-					d.ForceNew("sku")
-				} else {
-					log.Printf("[DEBUG] databricks workspace can be upgraded from %s to %s", oldSku, newSku)
+				if (customerEncryptionEnabled.(bool) || defaultStorageFirewallEnabled.(bool) || len(enhancedSecurityCompliance.([]interface{})) > 0 || infrastructureEncryptionEnabled.(bool) || managedServicesCMK.(string) != "" || managedDiskCMK.(string) != "") && !strings.EqualFold("premium", newSku.(string)) {
+					return fmt.Errorf("'customer_managed_key_enabled', 'default_storage_firewall_enabled', 'enhanced_security_compliance', 'infrastructure_encryption_enabled', 'managed_disk_cmk_key_vault_key_id' and 'managed_services_cmk_key_vault_key_id' are only available with a 'premium' workspace 'sku', got %q", newSku)
 				}
-			}
 
-			if (customerEncryptionEnabled.(bool) || defaultStorageFirewallEnabled.(bool) || infrastructureEncryptionEnabled.(bool) || managedServicesCMK.(string) != "" || managedDiskCMK.(string) != "") && !strings.EqualFold("premium", newSku.(string)) {
-				return fmt.Errorf("'customer_managed_key_enabled', 'default_storage_firewall_enabled', 'infrastructure_encryption_enabled', 'managed_disk_cmk_key_vault_key_id' and 'managed_services_cmk_key_vault_key_id' are only available with a 'premium' workspace 'sku', got %q", newSku)
-			}
+				return nil
+			}),
 
-			return nil
-		}),
+			// Once compliance security profile has been enabled, disabling it will force a workspace replacement
+			pluginsdk.ForceNewIfChange("enhanced_security_compliance.0.compliance_security_profile_enabled", func(ctx context.Context, old, new, meta interface{}) bool {
+				return old.(bool) && !new.(bool)
+			}),
+
+			// Once a compliance standard is enabled, disabling it will force a workspace replacement
+			pluginsdk.ForceNewIfChange("enhanced_security_compliance.0.compliance_security_profile_standards", func(ctx context.Context, old, new, meta interface{}) bool {
+				removedStandards := old.(*pluginsdk.Set).Difference(new.(*pluginsdk.Set))
+				return removedStandards.Len() > 0
+			}),
+
+			// Compliance security profile requires automatic cluster update and enhanced security monitoring to be enabled
+			pluginsdk.CustomizeDiffShim(func(ctx context.Context, d *pluginsdk.ResourceDiff, v interface{}) error {
+				_, complianceSecurityProfileEnabled := d.GetChange("enhanced_security_compliance.0.compliance_security_profile_enabled")
+				_, automaticClusterUpdateEnabled := d.GetChange("enhanced_security_compliance.0.automatic_cluster_update_enabled")
+				_, enhancedSecurityMonitoringEnabled := d.GetChange("enhanced_security_compliance.0.enhanced_security_monitoring_enabled")
+
+				if complianceSecurityProfileEnabled.(bool) && (!automaticClusterUpdateEnabled.(bool) || !enhancedSecurityMonitoringEnabled.(bool)) {
+					return fmt.Errorf("'automatic_cluster_update_enabled' and 'enhanced_security_monitoring_enabled' must be set to true when using 'compliance_security_profile'")
```

Comment:
We're referring to a property or block called `compliance_security_profile` but I don't see that in the schema, should this be 
```suggestion
					return fmt.Errorf("'automatic_cluster_update_enabled' and 'enhanced_security_monitoring_enabled' must be set to true when using 'compliance_security_profile_standards'")
```

Reply:
Fixed in https://github.com/hashicorp/terraform-provider-azurerm/commit/0159fa6fbef247bf18d40d1a9c9ec80247582a46